wiki_articles/std_array.md

```diff
@@ -0,0 +1,33 @@
+# What Is `std::array` And Why Should I Use It?
+
+C-style arrays have a few properties that make them annoying to use. They:
+
+:no_entry: cannot be returned from functions<br>
+:no_entry: are non-assignable<br>
+:no_entry: can only be initialized via `""`, `{}`, or *[default initialization](https://en.cppreference.com/w/cpp/language/default_initialization)*<br>
+:no_entry: cannot be passed to functions by value, but *[decay](https://64.github.io/cpp-faq/decay/)* to pointers<br>
+:warning: might implicitly turn into pointers, which is called *[array-to-pointer conversion](https://en.cppreference.com/w/cpp/language/implicit_conversion#Array-to-pointer_conversion)*<br>
```

Comment:
It's redundant to mention decay in both of these bullet points

Reply:
No, it's not, *decay* and *array-to-pointer* conversion are two different things, although people commonly confuse them and misuse these terms interchangeably. There are actually three separate effects at play:

1. [Type adjustment](https://eel.is/c++draft/dcl.fct#5) of "array of `T`" to "pointer of `T`".
2. [*Decay*](https://eel.is/c++draft/meta.trans.other#tab:meta.trans.other-row-4) is the change in argument types when calling functions.
3. [*Array-to-pointer conversion*](https://eel.is/c++draft/conv.array#1) which is more broad, and lets you do things like `array[i]` even though arrays don't have subscript operator.

Reply:
I understand what you're trying to do, however even if there was a distinction between decay and array-to-pointer conversion it's not suited for a beginner-oriented article.

>  2. [_Decay_](https://eel.is/c++draft/meta.trans.other#tab:meta.trans.other-row-4), the change in argument types when calling functions.

Array-to-pointer conversion is actually still what happens to arguments when calling a function, http://eel.is/c++draft/over#ics.scs-3, there's no distinction with "decay" there. The reason std::decay is called std::decay is because that's the widely adopted terminology for this behavior.